scripts/src/main/resources/scripts/environment-project.bat

```diff
@@ -84,29 +84,25 @@ set "line=%*"
 if "%~1%" == "export" (
   rem remove export as this only makes sense on linux/mac
   set line=!line:~7!
-  if "%~3%" == "" (
-    set "search=%~2%"
-  ) else (
-    set "search=%~2% "
-  )
-  set "replacement=%~2%="
+  shift
+)
+if "%~2%" == "" (
+  set "search=%~1%"
 ) else (
-  if "%~2%" == "" (
-    set "search=%~1%"
-  ) else (
-    set "search=%~1% "
-  )
-  set "replacement=%~1%="
+  set "search=%~1% "
+)
+set "replacement=%~1%="
+set value=%~2
+if "!value:~0,1!" == "~" (
+  rem variable value starts with tilde that needs to be replaced with users home dir (USERPROFILE)
+  set line=%replacement%%USERPROFILE%!value:~1!
```

Comment:
for me it's fine to just support ~ replacement at the first character of the value. Anyhow just in case you havent seen it, there are solutions to a more generic algorithm, which anyhow underlines how crappy batch is :( https://stackoverrun.com/de/q/9716479

Reply:
Thanks for the link. After rethinking it might even be wrong to resolve `C:/data/file~name` to `C:/data/fileC:\users\hohwillename`. So I would rather consider this as a feature :)